x/gamm/types/msg_sybil.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// SybilResistantFee Swap Msg defines an interface for getting
+// - poolIds on a swap msg route
+// - token to apply the sybil resistant fees to
+type SybilResistantFee interface {
+	GetTokenDenomsOnPath() []string
+	GetPoolIdOnPath() []uint64
+	GetTokenToFee() sdk.Coin
+}
```

Comment:
I think we should just add the `GetPoolIdOnPath` to the existing `msg_swap.go` interface. Not sure why "GetTokenToFee" exists

Reply:
By using a separate interface, `SybilResistantFee` can be more abstractly applied to any message involving a swap, whereas `SwapMsgRoute` is only satisfied by messages with swap routes in their proto file.
[These lines](https://github.com/osmosis-labs/osmosis/blob/d7fa2fd172668b4a0eb7d8207b3e2d0c952d43b6/x/txfees/keeper/feedecorator.go#L167-L185) of `feedecorator.go` use this abstraction and could work for msgs like `MsgJoinSwapExternAmountIn` properly satisfying `SybilResistantFee`. 
`GetTokenToFee()` returns the `sdk.Coin` that would have paid the swap fees, in its initial amount. This is then used to check if the token is applicable for use as a `feeToken`, and to compute the fees paid in the base denom.
If the swap is a `multihop`, then only swap fees from pools with a `feeToken` are added to the total swap fee.

Reply:
If you only see this as being used for swaps, not joins and exits, I'll move the new functionality to the existing interface.